### ISSUE TITLE
Notices do not get displayed during runtime

### ISSUE THREAD

**Issue by andywicht** [serge-rider/dbeaver#1298]
I have tested this only with a PostgreSQL connection using dbeaver 3.7.X, 3.8.X and 4.0.0.

The issue is that functions which raise notices (`raise notice 'test';`) during their runtime (e.g. to provide information about the progress) are not being displayed while the function runs. 
Once the function has finished the calculation, all notices are being displayed at once in the output tab.

In this case the behaviour of pgAdmin is desired: a notice is displayed the same moment it is raised by a function.

**Comment by andywicht** [comment]
okay thank you! that would be a great addition! especially in times where good alternatives to pgAdmin are crucial!

**Comment by serge-rider** [comment]
After all I think this is not possible with JDBC driver.
We going to add SQL debugger feature will will execute procedures on server-side step-by-step - in this case you will see all messages right in the moment they appear.

**Comment by andywicht** [comment]
Okay. Thank you for the investigation.
The debugging feature is great news!
Keep up the good work!

**Comment by GPFisher** [comment]
It appears that this is now working in v4.2.3!  That is great, as this is a very useful feature.

**Comment by antimirov** [comment]
It's not working for me with DBeaver 4.2.3 and PgJDBC 42.1.4. I see on pgjdbc's github page that there was a bug introduced at some point when the messages weren't sent as soon as they arrived, but only at the end. GPFisher, could you please tell me your version of pgjdbc? I tried to download some old version, but it also did work there with my DBeaver 4.2.3.

**Comment by GPFisher** [comment]
I checked again, and I realized that the function I was calling was much faster than I expected, so it looks like it is not happening in real-time after all. Bummer.

**Comment by serge-rider** [comment]
If you execute script on client side (<kbd>alt+x</kbd>) then you will see server-side messages once they will be produced by server.

But if you execute a single query (<kbd>ctrl+enter</kbd>)which produces multiple output then you'll see them only in the very end (because for DBeaver this operation is atomic and there is no special API for server messages reading).

So, this can be solved only by debugger (#1856)

**Comment by antimirov** [comment]
@serge-rider , how can I see the output when executing script on client side by Alt-X? When I press Alt-X, the Output View disappears and returns only after the script is fully executed.

**Comment by serge-rider** [comment]
You are right.

We going to improve existing output window (make it detachable/configurable).
For now as a workaround I'll add extra config option in 4.2.5:
![image](https://user-images.githubusercontent.com/6398845/32435833-32b56748-c2f3-11e7-90ea-a084391c2d96.png)

**Comment by antimirov** [comment]
Great, thanks!

**Comment by antimirov** [comment]
Just installed 4.2.5, still the same problem, even with this new Checkbox. Output view is now visible (btw, with and without this checkbox activated), but still I see the output only after this code finishes (10 seconds):

```
CREATE OR REPLACE FUNCTION test_print(t text) RETURNS integer AS $$
DECLARE
    i INTEGER;
BEGIN
	FOR i IN SELECT * FROM generate_series(1,1000) LOOP
	   RAISE NOTICE E't=\'%\' i=%\n', t, i;
	   PERFORM pg_sleep(0.01);
    END LOOP;	
    
RETURN i;
END;
$$ LANGUAGE plpgsql;

SELECT test_print('text');
```

**Comment by serge-rider** [comment]
Well, async messages pump is not supported by latest PG driver.
DBeaver will show messages after each script statement. 
For example:
```sql
...
SELECT test_print('text');
SELECT test_print('text2');
SELECT test_print('text3');
SELECT test_print('text4');
SELECT test_print('text5');
SELECT test_print('text6');
```
Messages will dumped after each SELECT query.

**Comment by antimirov** [comment]
So is this the correct PG driver bug - https://github.com/pgjdbc/pgjdbc/pull/857 ? I mean, once this is fixed, is it correct to assume even today's version of DBeaver will automatically start showing the output of test_print()?

Still, this is strange, actually, because a basic 'psql' tool has been showing messages for ages.

**Comment by antimirov** [comment]
Hi again,

As I understand, the https://github.com/pgjdbc/pgjdbc/pull/857 was merged into pgjdbc 42.2.0. But when I now try to use that driver (v42.2.1) with DBeaver 4.3.4 I don't see any changes at all. I run the same function SELECT test_print('text') but see the output only after 10 seconds, when it's fully done. Does this mean that that fix in JDBC driver was properly implemented, or rather that dbeaver does not enable this functionality by default for postgresql connections?

**Comment by serge-rider** [comment]
@antimirov Second. Currently DBeaver does't read server-side messages in async mode.
In this ticket we've added messages pump only for script mode (when you execute multiple queries).
I think we need another ticket for async messages reading - it is not too complicated to implement. But this ticket become too messy.

**Comment by 911zrfelix** [comment]
@serge-rider I compared pgAdmin4 and OminiDB with DBeaver, and I would say DBeaver is better than the other two in many aspects. I am not shift from pgadmin4 to dbeaver. We wish the next version can add the 'raise notice' function for PostgreSQL since this function is very useful with high and urgent demand for us. Without this function, it is really inconvenient for optimizing our codes. We need 'raise notice'. Thanks a lot for your work and devotion.

**Comment by 911zrfelix** [comment]
@serge-rider sorry. it should be "I shifted it from pgadmin4 to dbeaver".

**Comment by serge-rider** [comment]
We'll track this feature in #3322

**Comment by serge-rider** [comment]
Async server output reading will be added in 5.0.5 (just for PostgreSQL yet).
Please test it.

**Comment by antimirov** [comment]
Works as expected. Thank you!

**Comment by fernando-escobar** [comment]
I don't see it the output yet. Is there something I have to toggle?

**Comment by programmeravi** [comment]
> Just installed 4.2.5, still the same problem, even with this new Checkbox. Output view is now visible (btw, with and without this checkbox activated), but still I see the output only after this code finishes (10 seconds):
> 
> ```sql
> CREATE OR REPLACE FUNCTION test_print(t text) RETURNS integer AS $$
> DECLARE
>     i INTEGER;
> BEGIN
> 	FOR i IN SELECT * FROM generate_series(1,1000) LOOP
> 	   RAISE NOTICE E't=\'%\' i=%\n', t, i;
> 	   PERFORM pg_sleep(0.01);
>     END LOOP;	
>     
> RETURN i;
> END;
> $$ LANGUAGE plpgsql;
> 
> SELECT test_print('text');
> ```

HI I have tried the code mentioned above and it doesn't work as expected, 
My DBeaver version - Version 6.3.4.202002011957
